x-pack/legacy/plugins/uptime/server/lib/adapters/saved_objects/kibana_saved_objects_adapter.ts

```diff
@@ -22,4 +27,14 @@ export const savedObjectsAdapter: UMSavedObjectsAdapter = {
       );
     }
   },
+  getUptimeSourceSettings: async client => {
+    try {
+      return await client.get(
+        uptimeSourceConfigurationSavedObjectType,
+        uptimeSourceConfigurationSavedObjectId
+      );
+    } catch (e) {
+      return await client.create(uptimeSourceConfigurationSavedObjectType, defaultSourceSettings);
```

Comment:
unless `e` is a not found error, you probably want to re-throw e, otherwise the true error would be masked by a potential create error. You can test with https://github.com/elastic/kibana/blob/260d907f90173646498f927b0866279ce5a0fe73/src/core/server/saved_objects/service/lib/errors.ts#L147

Same thing in `getUptimeIndexPattern` above.

Reply:
Thanks! Fixed